src/handlers/sign.c

```diff
@@ -168,6 +168,17 @@ int handleSign(uint8_t p1, uint8_t p2, uint8_t *workBuffer, uint16_t dataLength)
 
     data_warning = ((txContent.dataBytes > 0) ? true : false);
 
+#ifdef HAVE_SWAP
+    if (G_called_from_swap) {
+        if ((txContent.contractType != TRANSFERCONTRACT) &&
+            (txContent.contractType != TRANSFERASSETCONTRACT) &&
+            (txContent.contractType != TRIGGERSMARTCONTRACT)) {
```

Comment:
Can't we restrict this more ?

Reply:
I've restricted it a bit more, but not much as this is what is needed from product side.